src/bot.ts

```diff
@@ -0,0 +1,107 @@
+import { Issue, IssueComment } from "@octokit/webhooks-types";
+
+import { CommentsApi } from "./github/comments";
+import { Merger } from "./github/merger";
+import { ActionLogger } from "./github/types";
+
+const BOT_COMMAND = "/bot";
+
+type Command = "merge" | "cancel" | "help";
+
+const botCommands = `
+**Available commands**
+
+- \`/bot merge\`: Enables auto-merge for Pull Request
+- \`/bot cancel\`: Cancels auto-merge for Pull Request
+- \`/bot help\`: Shows this menu
+
+For more information see the [documentation](https://github.com/paritytech/auto-merge-bot)
+`;
+
+export class Bot {
+  constructor(
+    private readonly comment: IssueComment,
+    private readonly pr: Issue,
+    private readonly logger: ActionLogger,
+    private readonly commentsApi: CommentsApi,
+  ) {}
+
+  /** Verifies if the author is the also the author of the PR or a member of the org */
+  async canTriggerBot(): Promise<boolean> {
+    this.logger.debug("Evaluating if user can trigger the bot");
+    const author = this.pr.user.id;
+    if (this.comment.user.id === author) {
+      this.logger.debug("Author of comment is also author of PR");
+      return true;
+    }
+    this.logger.debug("Author of comment is not the author of the PR");
+
+    return await this.commentsApi.userBelongsToOrg(this.comment.user.login);
+  }
+
+  async run(merger: Merger): Promise<void> {
+    this.logger.info("Running action on comment: " + this.comment.html_url);
+    if (!this.comment.body.startsWith(BOT_COMMAND)) {
```

Comment:
How do you feel about moving this check to the action yml, like here:

https://github.com/paritytech/rfc-action/blob/be7c934ade5bfce5eae889db3d6a286560431333/.github/workflows/rfc-action.yml#L13

The upside would be that we wouldn't unnecessarily start the program only to skip it, and also in the list of action runs we would clearly see which are skipped and which are run, like here: https://github.com/polkadot-fellows/RFCs/actions/workflows/rfc-action.yml

Reply:
Sounds like a good idea, but I would keep it anyway in the code as that configuration is set by the user, not the application (and the user may forget).